packages/blade/src/hooks/useSafeLayoutEffect.ts

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import { getPlatformType } from '~utils';
+
+const isBrowser = getPlatformType() == 'browser';
+
+/**
+ * useSafeLayoutEffect enables us to safely call `useLayoutEffect` on the browser
+ * (for SSR reasons)
+ *
+ * React currently throws a warning when using useLayoutEffect on the server.
+ * To get around it, we can conditionally useEffect on the server (no-op) and
+ * useLayoutEffect in the browser.
+ *
+ * @see https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85
+ */
+export const useSafeLayoutEffect = isBrowser ? React.useLayoutEffect : React.useEffect;
```

Comment:
Again we need it for the `useId` hook which is taken from https://reach.tech/auto-id/

I assume it requires layout effect to generate the pesudo-random ID before any painting happens on the users screen since if we do it after rendering there will be a moment when the React component won't have any ID associated with it thus it will break screen readers.

Reply:
This is Confirmed from the source comment. we need useLayoutEffect.
https://github.com/reach/reach-ui/blob/d9253130f9255dc8e27a0edd3ce8e6943a444356/packages/auto-id/src/auto-id.ts#L104-L109